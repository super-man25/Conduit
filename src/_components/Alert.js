// @flow
import * as React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import styled from 'styled-components';

import { colors, shadows, zIndexes } from '_constants';
import { actions } from '_state/alert';
import { Icon } from './Icon';
import { Flex } from './Flex';

const StyledApiAlert: React.ComponentType<{}> = styled(Flex)`
  position: absolute;
  right: 25px;
  top: 25px;
  z-index: ${zIndexes.BASE};
  width: 350px;
  background-color: ${colors.white};
  border-color: ${(props) =>
    props.type === 'apiError' ? colors.red : colors.green};
  box-shadow: ${shadows.SMALL};
  visibility: ${(props) => (props.show ? 'visible' : 'hidden')};
  cursor: pointer;
  padding: 25px;
`;

const StatusIcon: React.ComponentType<{}> = styled.div`
  margin-right: 15px;
`;

const StatusText: React.ComponentType<{}> = styled.div`
  font-size: 20px;
  font-weight: 300;
  letter-spacing: 1px;
  color: ${(props) => props.color};
  margin-bottom: 3px;
`;

const MsgText: React.ComponentType<{}> = styled.div`
  font-size: 13px;
  font-weight: 300;
`;

export const Alert = () => {
  const alertState = useSelector(({ alert }) => alert);
  const dispatch = useDispatch();
  const clearAlert = () => dispatch(actions.clear());

  const show = alertState.type !== null && alertState.message !== null;
  const statusColor =
    alertState.type === 'apiError' ? colors.red : colors.green;
  const statusText = alertState.type === 'apiError' ? 'Error:' : 'Success!';

  return (
    <StyledApiAlert {...alertState} show={show} onClick={clearAlert}>
      <StatusIcon>
        <Icon name={alertState.type} size={40} color={statusColor} />
      </StatusIcon>
      <Flex direction="column">
        <StatusText color={statusColor}>{statusText}</StatusText>
        <MsgText>{alertState.message}</MsgText>
      </Flex>
    </StyledApiAlert>
  );
};
