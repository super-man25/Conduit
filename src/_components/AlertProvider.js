import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { history } from '_helpers';
import { actions as alertActions } from '_state/alert';

class AlertProviderPresenter extends React.Component {
  constructor(props) {
    super(props);

    const { dispatch } = this.props;
    history.listen(() => {
      dispatch(alertActions.clear());
    });
  }

  render() {
    const { alert, children } = this.props;

    // TODO: style alerts
    return (
      <div>
        {alert &&
          alert.show && (
            <div className={`alert ${alert.type}`}>{alert.message}</div>
          )}
        {children}
      </div>
    );
  }
}

AlertProviderPresenter.propTypes = {
  /** Redux dispatch */
  dispatch: PropTypes.func,

  /** Alert object */
  alert: PropTypes.shape({
    message: PropTypes.string,
    type: PropTypes.string
  }),

  /** Children */
  children: PropTypes.node
};

function mapStateToProps(state) {
  return {
    alertState: state.alert
  };
}

export const AlertProvider = connect(mapStateToProps)(AlertProviderPresenter);
