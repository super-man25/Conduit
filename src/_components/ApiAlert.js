// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { cssConstants, shadows, zIndexes } from '_constants';
import { Icon } from './Icon';

type Props = {
  alertState: {
    type: 'api-error' | 'api-success',
    message: string
  }
};

export const ApiAlertDiv: React.ComponentType<{}> = styled.div`
  position: absolute;
  right: 24px;
  top: 24px;
  z-index: ${zIndexes.BASE};
  font-size: 18px;
  width: 357px;
  height: 90px;
  background-color: ${cssConstants.PRIMARY_WHITE};
  border-color: ${(props) =>
    props.type === 'api-error'
      ? cssConstants.SECONDARY_RED
      : cssConstants.SECONDARY_GREEN};
  box-shadow: ${shadows.SMALL};
  visibility: ${(props) => (props.show ? 'visible' : 'hidden')};
`;

const StatusIcon: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 26px;
  left: 26px;
`;

const StatusText: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 26px;
  left: 93px;
  font-size: 1.2em;
  font-weight: 300;
  letter-spacing: 1px;
  color: ${(props) => props.color};
`;

const MsgText: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 54px;
  left: 93px;
  font-size: 0.7em;
  font-weight: 300;
  margin-right: 24px;
  color: ${cssConstants.PRIMARY_BLACK};
`;

export class ApiAlertPresenter extends React.Component<Props> {
  render() {
    const { alertState } = this.props;
    const show = {
      show: alertState.type !== null && alertState.message !== null
    };
    const statusColor =
      alertState.type === 'api-error'
        ? cssConstants.SECONDARY_RED
        : cssConstants.SECONDARY_GREEN;
    const statusText = alertState.type === 'api-error' ? 'Error:' : 'Success!';

    return (
      <ApiAlertDiv {...alertState} {...show}>
        <StatusIcon>
          <Icon name={alertState.type} size={39} color={statusColor} />
        </StatusIcon>
        <StatusText color={statusColor}>{statusText}</StatusText>
        <MsgText>{alertState.message}</MsgText>
      </ApiAlertDiv>
    );
  }
}

function mapStateToProps(state) {
  return {
    alertState: state.alert
  };
}

export const ApiAlert = connect(
  mapStateToProps,
  null
)(ApiAlertPresenter);
