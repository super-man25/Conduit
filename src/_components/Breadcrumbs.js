import styled from 'styled-components';
import { cssConstants } from '../_constants';

export const Breadcrumbs = styled.div`
  width: 100%;
  height: 30px;
  margin: 0;
  padding: 0;
  background: ${cssConstants.PRIMARY_WHITE};
  font-size: 12px;
`;
