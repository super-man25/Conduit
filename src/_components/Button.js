// @flow

import * as React from 'react';
import styled from 'styled-components';
import { cssConstants, shadows } from '_constants';
import { withBoxModelProps } from '_helpers/style-utils';

type Props = {
  fontSize: string,
  textAlign: string,
  minWidth: string,
};

export const Button: React.ComponentType<Props> = withBoxModelProps(styled.button`
  border: none;
  min-width: ${(props) => props.minWidth || '100px'};
  padding: 1rem 2rem;
  text-align: ${(props) => props.textAlign || 'center'};
  border-radius: 3px;
  font-size: ${(props) => props.fontSize || '1rem'};
  cursor: pointer;
  transition-duration: 0.3s;

  &:disabled {
    cursor: not-allowed;
  }
`);

export const PrimaryButton = styled(Button)`
  transition: opacity 100ms ease-out;
  color: ${cssConstants.PRIMARY_WHITE};
  background-color: ${cssConstants.PRIMARY_BLUE};

  &:disabled {
    color: ${cssConstants.PRIMARY_GRAY};
    background-color: ${cssConstants.PRIMARY_LIGHT_GRAY};
    opacity: 0.5;
  }

  &:active {
    background-color: ${cssConstants.PRIMARY_LIGHT_BLUE};
    box-shadow: ${shadows.ACTIVE_BUTTON};
  }

  &:hover:not(:disabled) {
    background-color: ${cssConstants.PRIMARY_BLUE_HOVER};
  }
`;

export const SecondaryButton = styled(Button)`
  background-color: ${cssConstants.PRIMARY_WHITE};
  border: 2px solid ${cssConstants.PRIMARY_BLUE};
  color: ${cssConstants.PRIMARY_BLUE};
  transition: border linear;

  &:disabled {
    border: 2px solid ${cssConstants.PRIMARY_GRAY};
    color: ${cssConstants.PRIMARY_GRAY};
    opacity: 0.5;
  }

  &:active {
    background-color: ${cssConstants.SECONDARY_BACKGROUND_LIGHTEST_BLUE};
    color: ${cssConstants.PRIMARY_LIGHT_BLUE};
    outline: 2px solid ${cssConstants.PRIMARY_LIGHT_BLUE};
    box-shadow: ${shadows.ACTIVE_BUTTON};
  }

  &:hover:not(:disabled) {
    border: 3px solid ${cssConstants.PRIMARY_BLUE};
    background-color: ${cssConstants.SECONDARY_BACKGROUND_LIGHTEST_BLUE};
    color: ${cssConstants.PRIMARY_BLUE};
  }
`;

export const TertiaryButton = styled(Button)`
  background-color: ${cssConstants.PRIMARY_LIGHT_GRAY};
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
  opacity: 0.8;

  &:disabled {
    opacity: 0.5;
  }

  &:hover:not(:disabled) {
    opacity: 1;
    background-color: ${cssConstants.PRIMARY_LIGHT_GRAY};
  }
`;

export const TextButton = styled(Button)`
  color: ${cssConstants.PRIMARY_BLUE};
  background-color: inherit;

  &:disabled {
    color: ${cssConstants.PRIMARY_LIGHT_GRAY};
    opacity: 0.5;
  }

  &:hover:not(:disabled) {
    color: ${cssConstants.PRIMARY_LIGHT_BLUE};
  }
`;
