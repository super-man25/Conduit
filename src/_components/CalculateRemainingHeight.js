import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { ReactHeight } from 'react-height';

const Container = styled.div`
  height: 100%;
`;

export class CalculateRemainingHeight extends React.Component {
  constructor(props) {
    super(props);
    this.containerRef = React.createRef();
  }

  state = {
    remainingHeight: null
  };

  setRemainingHeight = (contentHeight) => {
    const containerNode = this.containerRef.current;
    this.setState({
      remainingHeight: containerNode.clientHeight - contentHeight
    });
  };

  render() {
    const children = [].concat(this.props.children);
    const { remainingHeight } = this.state;

    const total = children.length;

    const nodes = children.slice(0, total - 1);
    const func = children[total - 1];

    if (typeof func !== 'function') {
      throw new Error(
        'CalculateRemainingHeight: Expected the last child to be a function'
      );
    }

    return (
      <Container innerRef={this.containerRef}>
        <ReactHeight onHeightReady={this.setRemainingHeight}>
          {nodes}
        </ReactHeight>
        {remainingHeight !== 0 && func(remainingHeight)}
      </Container>
    );
  }
}

const childProp = PropTypes.oneOfType([PropTypes.node, PropTypes.func]);

CalculateRemainingHeight.propTypes = {
  /**
   * Children to display. The last child should be a function which receives the remaining
   * height in the container as a parameter and returns a node to render.
   */
  children: PropTypes.oneOfType([childProp, PropTypes.arrayOf(childProp)])
};
