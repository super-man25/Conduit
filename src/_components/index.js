export * from './Alert';
export * from './AsyncButton';
export * from './Breadcrumbs';
export * from './Button';
export * from './Center';
export * from './CenteredLoader';
export * from './ChartContainer';
export * from './ChartLegend';
export * from './Checkbox';
export * from './ChipButton';
export * from './ContentWrapper';
export * from './DatePicker';
export * from './DateRangeDropdown';
export * from './DateRangeFilter';
export * from './Dropdown';
export * from './DropdownButton';
export * from './EDLink';
export * from './EDText';
export * from './Flex';
export * from './Form';
export * from './FormInput';
export * from './FormInputArray';
export * from './HelpBlockDiv';
export * from './HorizontalBars';
export * from './Icon';
export * from './Icon/SVGIcons';
export * from './ImageLayout';
export * from './Input';
export * from './Integration';
export * from './Layout';
export * from './LeftNav';
export * from './LegendItem';
export * from './Loader';
export * from './MailtoLink';
export * from './MainContent';
export * from './Modal';
export * from './Panel';
export * from './ReportDownloadButton';
export * from './ScreenTitleBlock';
export * from './ScrollableList';
export * from './SecuredRoute';
export * from './SelectBox';
export * from './SelectDropdown';
export * from './Setting';
export * from './SettingButton';
export * from './SiteHeader';
export * from './SortableButton';
export * from './Spacing';
export * from './StyledTags';
export * from './Table';
export * from './Toggle';
export * from './UserWelcome';
