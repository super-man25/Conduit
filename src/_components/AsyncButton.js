// @flow

import * as React from 'react';
import { cssConstants } from '_constants';
import { PrimaryButton, Loader } from '_components';

type Props = {
  isLoading: boolean,
  disabled?: boolean,
  children: React.Node,
};

export const AsyncButton = (props: Props) => {
  const { isLoading, children, ...rest } = props;
  return (
    <PrimaryButton {...rest}>
      {props.isLoading ? (
        <Loader small color={cssConstants.PRIMARY_WHITE} />
      ) : (
        props.children
      )}
    </PrimaryButton>
  );
};
