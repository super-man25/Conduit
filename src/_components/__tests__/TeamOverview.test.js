import React from 'react';
import { TeamOverviewPresenter as TeamOverview } from '_components';
import renderer from 'react-test-renderer';

const createProps = (update) => ({
  stats: { wins: 10, losses: 1, ties: 3, gamesTotal: 14 },
  onToggleSidebar: jest.fn(),
  seasons: [],
  selectedSeason: null,
  setActiveSeasonId: jest.fn(),
  eventList: [],
  client: {
    performanceType: 'MLB',
  },
  ...update,
});

it('renders correctly with MLB performance type', () => {
  const props = createProps();
  const tree = renderer.create(<TeamOverview {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with NFL performance type', () => {
  const props = createProps({
    client: {
      performanceType: 'NFL',
    },
  });
  const tree = renderer.create(<TeamOverview {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with MLS performance type', () => {
  const props = createProps({
    client: {
      performanceType: 'MLS',
    },
  });
  const tree = renderer.create(<TeamOverview {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with NCAAF performance type', () => {
  const props = createProps({
    client: {
      performanceType: 'NCAAF',
    },
  });
  const tree = renderer.create(<TeamOverview {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with NCAAB performance type', () => {
  const props = createProps({
    client: {
      performanceType: 'NCAAB',
    },
  });
  const tree = renderer.create(<TeamOverview {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with toggledSidebar', () => {
  const props = createProps();
  const tree = renderer.create(<TeamOverview {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
