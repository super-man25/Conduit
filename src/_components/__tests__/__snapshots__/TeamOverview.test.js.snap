// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with props 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
}

.c4 path {
  fill: #FFFFFF;
}

.c6 {
  display: block;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 200;
  width: 30px;
  margin: 0;
  margin-top: 20px;
  margin-bottom: auto;
  margin-right: 20px;
  padding: 0.4em;
  padding-top: 0.3em;
  padding-bottom: 0.3em;
  border: 2px solid;
  border-radius: 3px;
  border-color: #38A9DB;
  background-color: #38A9DB;
}

.c6:hover {
  cursor: pointer;
  color: #FFFFFF;
  background-color: #2670AE;
  border-color: #2670AE;
}

.c6::before {
  content: ' 























































.c1 {;
}

.c2 {
  position: relative;
  cursor: pointer;
}

.c5 {
  position: absolute;
  width: 180px;
  top: calc(100% + 8px);
  z-index: 10;
  border: 1px solid #EEEEEE;
  box-shadow: 0 20px 20px rgba(0,0,0,0.06);
  -webkit-transition: 0.1s ease-in-out all;
  transition: 0.1s ease-in-out all;
  opacity: 0;
  -webkit-transform: translateY(20px);
  -ms-transform: translateY(20px);
  transform: translateY(20px);
  visibility: hidden;
}

.c0 {
  height: 100%;
  width: auto;
  margin: 0;
  padding-left: 20px;
  color: #FFFFFF;
  background: 'none';
}

.c3 {
  color: #070707;
  font-size: 18px;
  font-weight: normal;
  margin: 0;
  padding: 0;
  vertical-align: top;
  color: #FFFFFF;
}

.c7 {
  color: #024A8F;
  font-size: 22px;
  font-weight: normal;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  weight: 300;
}

<div
  className="c0"
>
  <div
    className=""
    direction="row"
  >
    <div>
      <div
        className="c1 c2"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <h4
            className="c3"
            color="#070707"
          >
            Select a Season
          </h4>
          <svg
            className="c4"
            color="#FFFFFF"
            viewBox="0 0 24 24"
            width="24px"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <div
          className="c5"
        />
      </div>
    </div>
    <button
      className="c6"
      onClick={[MockFunction]}
    />
  </div>
  <div
    className=""
    direction="row"
  >
    <div
      className=""
    >
      <h3
        className="c7"
        color="#024A8F"
      >
        0.909
      </h3>
      <p
        className=""
        color="#FFFFFF"
      >
        <i>
          Win / Loss
        </i>
      </p>
    </div>
    <div
      className=""
    >
      <h3
        className="c7"
        color="#024A8F"
      >
        -10
      </h3>
      <p
        className=""
        color="#FFFFFF"
      >
        <i>
          Games Remaining
        </i>
      </p>
    </div>
  </div>
</div>
`;

exports[`renders correctly with toggledSidebar 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
}

.c4 path {
  fill: #FFFFFF;
}

.c6 {
  display: block;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 200;
  width: 30px;
  margin: 0;
  margin-top: 20px;
  margin-bottom: auto;
  margin-right: 20px;
  padding: 0.4em;
  padding-top: 0.3em;
  padding-bottom: 0.3em;
  border: 2px solid;
  border-radius: 3px;
  border-color: #38A9DB;
  background-color: #38A9DB;
}

.c6:hover {
  cursor: pointer;
  color: #FFFFFF;
  background-color: #2670AE;
  border-color: #2670AE;
}

.c6::before {
  content: ' 























































.c1 {;
}

.c2 {
  position: relative;
  cursor: pointer;
}

.c5 {
  position: absolute;
  width: 180px;
  top: calc(100% + 8px);
  z-index: 10;
  border: 1px solid #EEEEEE;
  box-shadow: 0 20px 20px rgba(0,0,0,0.06);
  -webkit-transition: 0.1s ease-in-out all;
  transition: 0.1s ease-in-out all;
  opacity: 0;
  -webkit-transform: translateY(20px);
  -ms-transform: translateY(20px);
  transform: translateY(20px);
  visibility: hidden;
}

.c0 {
  height: 100%;
  width: auto;
  margin: 0;
  padding-left: 20px;
  color: #FFFFFF;
  background: 'none';
}

.c3 {
  color: #070707;
  font-size: 18px;
  font-weight: normal;
  margin: 0;
  padding: 0;
  vertical-align: top;
  color: #FFFFFF;
}

.c7 {
  color: #024A8F;
  font-size: 22px;
  font-weight: normal;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  weight: 300;
}

<div
  className="c0"
>
  <div
    className=""
    direction="row"
  >
    <div>
      <div
        className="c1 c2"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <h4
            className="c3"
            color="#070707"
          >
            Select a Season
          </h4>
          <svg
            className="c4"
            color="#FFFFFF"
            viewBox="0 0 24 24"
            width="24px"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <div
          className="c5"
        />
      </div>
    </div>
    <button
      className="c6"
      onClick={[MockFunction]}
    />
  </div>
  <div
    className=""
    direction="row"
  >
    <div
      className=""
    >
      <h3
        className="c7"
        color="#024A8F"
      >
        0.909
      </h3>
      <p
        className=""
        color="#FFFFFF"
      >
        <i>
          Win / Loss
        </i>
      </p>
    </div>
    <div
      className=""
    >
      <h3
        className="c7"
        color="#024A8F"
      >
        -10
      </h3>
      <p
        className=""
        color="#FFFFFF"
      >
        <i>
          Games Remaining
        </i>
      </p>
    </div>
  </div>
</div>
`;
